### DISCUSSION TITLE
Block Mode for conditional statements?

### DISCUSSION THREAD

**Discussion by RogerChristopher** [STweep/STweep#72] [Ideas]
Hi Gerhard

Can we have a "block mode" for lengthy expressions in conditional statements? One idea for this would be to put long expressions in parentheses to switch this feature on (as we would not want it for all IF statements). For Example, the following code:
```
IF (Length_Expression) THEN
    Statements...
END_IF
```
would be formatted as:
```
IF 
    (
    Lengthy_Expression
    )
THEN
    Statements...
END_IF
```
The Lengthy_Expression would not get re-formatted by STweep so the developer can arrange the expression on multiple lines for maximum clarity. 

Omitting the parentheses just results in a regular IF statement:
```
IF Lengthy_Expression THEN
    Statements...
END_IF
```
I know you should limit the complexity of conditional expressions, but they still creep in.

Regards

Roger

**Comment by Barteling** [comment]
Hi Roger,

As it happens i'm just finalizing a new beta which has this feature. 
More or less then. 

So I suggest we continue discussion here after you reviewed that.

It can do wat you say, but  then it is not triggered by parentheses but by line length overflow or multi line

For example

```
IF veryLongStatementWhichTriggersLineLengthOverflowOrIsMultiline THEN
;
END_IF
```

can be formatted to
```
IF
     veryLongStatementWhichTriggersLineLengthOverflowOrIsMultiline
THEN
;
END_IF
```

New options:
![image](https://user-images.githubusercontent.com/33071638/191113071-9f7b9ecd-3813-45ab-8a58-b25ead8497ec.png)

**Comment by RogerChristopher** [comment]
Ok, sounds good.

**Comment by Barteling** [comment]
Could you check the new behaviours in 4.1.0-Beta?
Guess this comes close to  what you want


https://www.stweep.com/beta-releases/

**Comment by RogerChristopher** [comment]
Hi Gerhard

This looks good, but could we have an option to not indent the THEN keyword? I think this separates the code blocks into a more readable pattern:



Regards

Roger

**Comment by Barteling** [comment]
Roger, please give the new search function a try ^^ haha

![image](https://user-images.githubusercontent.com/33071638/192511694-a010b967-24fa-4720-a7d7-392f82d70cb0.png)

**Comment by RogerChristopher** [comment]
Perfect!
